src/openrct2/paint/Painter.cpp

```diff
@@ -197,15 +195,13 @@ void Painter::ReleaseSession(PaintSession* session)
 {
     PROFILED_FUNCTION();
 
-    session->PaintEntryChain.Clear();
+    session->paintEntries.clear();
+
     _freePaintSessions.push_back(session);
 }
 
 Painter::~Painter()
 {
-    for (auto&& session : _paintSessionPool)
-    {
-        ReleaseSession(session.get());
-    }
+    _paintSessionPool.clear();
     _paintSessionPool.clear();
```

Comment:
This calls the same function twice, is that intentional?

Reply:
Good catch, don't know how this came to be.